network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,454 @@
+            description = dict(type='str'),
+            snat=dict(type='str'),
+            default_persistence_profile=dict(type='str')
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec = argument_spec,
+        supports_check_mode=True
+    )
+
+    (server,user,password,state,partition,validate_certs) = f5_parse_arguments(module)
+    name = fq_name(partition,module.params['name'])
+    destination=module.params['destination']
+    port=module.params['port']
+    all_profiles=fq_list_names(partition,module.params['all_profiles'])
+    pool=fq_name(partition,module.params['pool'])
+    description = module.params['description']
+    snat = module.params['snat']
+    default_persistence_profile=fq_name(partition,module.params['default_persistence_profile'])
+
+    if 1 > port > 65535:
+        module.fail_json(msg="valid ports must be in range 1 - 65535")
+  
+    try:
+        api = bigip_api(server, user, password)
+        result = {'changed': False}  # default
+
+        if state == 'absent':
+                if not module.check_mode:
+                    if vs_exists(api,name):
+                        # hack to handle concurrent runs of module
+                        # pool might be gone before we actually remove
+                        try:
+                            vs_remove(api,name)
+                            result = {'changed' : True, 'deleted' : name }
+                        except bigsuds.OperationFailed, e:
+                            if "was not found" in str(e):
+                                result['changed']= False
+                            else:
+                                raise
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state == 'present':
+            update = False
+            if not vs_exists(api, name):
+                if (not destination) or (not port):
+                    module.fail_json(msg="both destination and port must be supplied to create a VS")
+                if not module.check_mode:
+                    # a bit of a hack to handle concurrent runs of this module.
+                    # even though we've checked the pool doesn't exist,
+                    # it may exist by the time we run create_pool().
+                    # this catches the exception and does something smart
+                    # about it!
+                    try:
+                        vs_create(api,name,destination,port,pool)
+                        set_profiles(api,name,all_profiles)
+                        set_snat(api,name,snat)
+                        set_description(api,name,description)
+                        set_default_persistence_profiles(api,name,default_persistence_profile)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                            raise Exception('Error on creating Virtual Server : %s' % e)
```

Comment:
Can we space this over so it is aligned under the exception like the rest of the code?